build_tools/docker/dockerfiles/base.Dockerfile

```diff
@@ -5,7 +5,13 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 # 20.04
-FROM ubuntu@sha256:b795f8e0caaaacad9859a9a38fe1c78154f8301fdaf0872eaf1520d66d9c0b98
+#FROM ubuntu@sha256:b795f8e0caaaacad9859a9a38fe1c78154f8301fdaf0872eaf1520d66d9c0b98
+# From https://hub.docker.com/r/arm64v8/ubuntu/
+#FROM ubuntu@sha256:144e6a778925a0c11c4cd9fe5fce1172e620f215b0410bb43e7fa41bbcfe4522
+# Why not use this (platform independent afaik):
+FROM ubuntu:20.04
```

Comment:
Would it be ok to use the generic tag? If not, is there a nice way in docker to switch on platform? If not, the only option I see is a separate base dockerfile, but I might be missing something...

Reply:
As a general policy, we try to pin versions to hashes rather than generic tags. Some of that comes from https://opensource.google/documentation/reference/github/services#actions 

> When using a third-party action (one not hosted in a Google-managed org), a fixed version of the action MUST be used by [specifying a specific commit](https://help.github.com/en/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsuses), rather than a branch like "master", or a tagged release, which can be overwritten by any maintainer of the action. Docker images should always be run at a fixed version rather than "latest".

I'd need to read up more on how the Ubuntu Docker images are configured to understand the details here...

Reply:
It's also because it otherwise makes it difficult to rebuild things. Specific tags give us some semblance of hermeticness in our containers, where someone rebuilding the container will get the same thing again.

Regarding whether this can be a switch, I think you can use an `ARG`: https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact. We'll need to update our docker scripting to accommodate the use of args in that case

Reply:
Thx for the explanations. I'll go for a specific SHA for now then. Is any Ubuntu 20.04 SHA from https://hub.docker.com/r/arm64v8/ubuntu/ ok? I couldn't find a 1:1 mapping to the current x86 build?

> I think you can use an ARG: https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact.

Cool, will look into it.

Reply:
Yeah any version is fine. I'd just take whatever the 20.04 tag currently points at

Reply:
I'll modify the base dockerfile to take a SHA as build-arg. The default value would be the existing x86 SHA. Then modify `build_tools/docker/manage_images.py` to pass the and Ubuntu SHA via `--build-arg`. Not the cleanest solution IMO, but it doesn't seem like there's a better way to do a conditional FROM in docker... Let be know if you know of a better pattern.